src/api/v1/db/oracledb/ingredients-dao.js

```diff
@@ -258,14 +230,40 @@ const updateIngredientById = async (body) => {
 
     if (result.rowsAffected === 0) return null;
 
-    const rawIngredient = convertOutBindsToRawIngredient(result.outBinds);
+    const rawIngredient = convertOutBindsToRawResource(result.outBinds);
     if (rawIngredient.id !== body.data.id) throw new Error('ID returned from database does not match input ID');
     return serializeIngredient(rawIngredient, `ingredients/${body.data.id}`);
   } finally {
     connection.close();
   }
 };
+
+/**
+ * check if a list of ingredient IDs exists in the database
+ *
+ * @param {string[]} ingredientIds a list of ingredient IDs to check in the database
+ * @returns {boolean} returns true if ingredient IDs are valid
+ */
+const checkIngredientsExist = async (ingredientIds) => {
+  const connection = await getConnection();
+  const bindParams = _.reduce(ingredientIds,
+    (binds, curId) => {
+      binds[`id${curId}`] = curId;
+      return binds;
+    }, {});
+  const query = `SELECT Count(*) AS "count" FROM INGREDIENTS WHERE ID IN (${_.keys(bindParams).map((key) => `:${key}`)})`;
```

Comment:
this query could return a false negative if someone passed a duplicate of the same relationship resource, like
```json
"data" : [
  { "type": "ingredient", "id": 1 },
  { "type": "ingredient", "id": 1 },
]
```

but I think that's supposed to be an error condition anyways so I'm not sure if it's an issue

Reply:
From our discussion, it looks like we'll want to handle a request like this with a 400. Hopefully this shouldn't be too difficult to implement as a general middleware.

Reply:
@ianbrown9475 is there a good place to look for docs on adding middleware?

Reply:
The [express-openapi](https://github.com/kogosoftwarellc/open-api/tree/master/packages/express-openapi#table-of-contents) table of contents has some details about different kinds of middleware and how they can be set up.